projects/lib/src/xboardengine.cpp

```diff
@@ -624,10 +627,14 @@ void XboardEngine::parseLine(const QString& line)
 			return;
 		}
 
-		Chess::Move move = board()->moveFromString(args);
+		// remove "..." of old format if necessary
+		int mark = (args.indexOf("..."));
+		const QString movestr = mark < 0 ? args : args.right(mark + 4);
```

Comment:
Ah, well spotted! This is an ugly mistake. It is hidden most of the time but promotions would call for trouble. As you wrote `mid(mark+4)` is what I intended to do. And also mark should always be zero in this branch of execution, so `mid(4)` should be the same.